### ISSUE TITLE
slim-cli (binary): Promisse rejection

### ISSUE THREAD

**Issue by 0xE282B0** [INSIDEM2M/slim#117]
When I install slim like this:
`yarn global add slim-cli`

and start `slim dev` I get the following error:

```
(node:271) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): dll polyfills
Module not found: Error: Can't resolve 'source-map-loader' in '/root/.slim/app'
resolve 'source-map-loader' in '/root/.slim/app'
  Parsed request is a module
  using description file: /root/.slim/app/package.json (relative path: .)
  after using description file: /root/.slim/app/package.json (relative path: .)
    resolve as module
      looking for modules in /usr/local/share/.config/yarn/global/node_modules/slim-cli/node_modules
        using description file: /usr/local/share/.config/yarn/global/node_modules/slim-cli/package.json (relative path: ./node_modules)
        after using description file: /usr/local/share/.config/yarn/global/node_modules/slim-cli/package.json (relative path: ./node_modules)
          using description file: /usr/local/share/.config/yarn/global/node_modules/slim-cli/package.json (relative path: ./node_modules/source-map-loader)
            as directory
              /usr/local/share/.config/yarn/global/node_modules/slim-cli/node_modules/source-map-loader doesn't exist
            no extension
              /usr/local/share/.config/yarn/global/node_modules/slim-cli/node_modules/source-map-loader doesn't exist
            .js
              /usr/local/share/.config/yarn/global/node_modules/slim-cli/node_modules/source-map-loader.js doesn't exist
            .ts
              /usr/local/share/.config/yarn/global/node_modules/slim-cli/node_modules/source-map-loader.ts doesn't exist
      looking for modules in /usr/local/share/.config/yarn/global/node_modules/slim-cli/dist/webpack/loaders
        using description file: /usr/local/share/.config/yarn/global/node_modules/slim-cli/package.json (relative path: ./dist/webpack/loaders)
        after using description file: /usr/local/share/.config/yarn/global/node_modules/slim-cli/package.json (relative path: ./dist/webpack/loaders)
          using description file: /usr/local/share/.config/yarn/global/node_modules/slim-cli/package.json (relative path: ./dist/webpack/loaders/source-map-loader)
            as directory
              /usr/local/share/.config/yarn/global/node_modules/slim-cli/dist/webpack/loaders/source-map-loader doesn't exist
            no extension
              /usr/local/share/.config/yarn/global/node_modules/slim-cli/dist/webpack/loaders/source-map-loader doesn't exist
            .js
              /usr/local/share/.config/yarn/global/node_modules/slim-cli/dist/webpack/loaders/source-map-loader.js doesn't exist
            .ts
              /usr/local/share/.config/yarn/global/node_modules/slim-cli/dist/webpack/loaders/source-map-loader.ts doesn't exist
```

**Comment by frederikschubert** [comment]
I think this has to do with the `yarn.lock` file being included in the `slim-cli` distribution.

**Comment by frederikschubert** [comment]
I will publish a new patch to test the fix.

**Comment by frederikschubert** [comment]
This error is not the `yarn.lock` file. 

The problem is that we only look for modules in the `node_modules` directory of `slim-cli`. If it is globally installed and `slim-cli` depends package `foo` that is not already in the global `node_modules` directory `foo` is installed on the same level as `slim-cli`. 

We need to change the `resolveLoader` and `resolve` options to include `node_modules` to which the `slim-cli` was possibly installed.

**Comment by frederikschubert** [comment]
This only happens with yarn. So the current workaround would be to use npm till a fix is found.

**Comment by frederikschubert** [comment]
Fixed when using `yarn@0.20.3`.
